### PR TITLE
Fixing roster updates

### PR DESCRIPTION
**Pull request by FriedrichAltheide**
Allow jid as roster source
#379

### CODE REVIEW

File: Monal/Classes/MLIQProcessor.m [hunk 1]
```diff
@@ -318,10 +318,9 @@ -(void) discoResult:(ParseIq *) iqNode {
 }
 
 -(void) rosterResult:(ParseIq *) iqNode {
-    
-    if(iqNode.from != nil
-      && iqNode.from != self.connection.identity.domain) {
-        DDLogError(@"invalid sender for roster. Rejecintg. ");
+    if(iqNode.from == nil && iqNode.from != self.connection.identity.jid
```
Comment (anurodhp) [inline Monal/Classes/MLIQProcessor.m]: should this be != nil

**Comment by anurodhp** [comment]
i cant remember the XEPs here should the roster come from the jid?

**Comment by FriedrichAltheide** [comment]
Yes, it should

**Comment by tmolitor-stud-tu** [comment]
@anurodhp see the examples here: https://xmpp.org/extensions/xep-0237.html#example-4

**Pull request merged**